### PR TITLE
Remove introspection in testool

### PR DESCRIPTION
**Pull request by adria0**
Since https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1072 is merged, this PR removes the opcode and geth trace errors introspection. Now `testool` just panics on unimplementeds, and the test runner catches the error message and tags the failing test as `Ignored`.

### Contents
- Removes the `unimplemented_opcodes` in the `Config.toml`
- Allows to specify no gas limits (`max_gas=0`).
- Removes the `tofix` test tags, now only tests that SIGKILL the binary are avoided (`sigkill` tag)
- Now cache file can be used with `report` option. If cache file is used only discard tests that has not been previously Ignored.
- Some minor tty output tweaks, like trimming too large failure descriptions in the the `print_tty` 
- Spellcheck

### CODE REVIEW

File: testool/README.md [hunk 1]
```diff
@@ -95,16 +95,19 @@ Sometimes there are some files or specific tests that we want to disable at all.
 When the command line parameter `--report` is defined, it automatically: 
 
 - After the execution, a two files are created in the `report` folder. They are
-   - `<timestamp>-<git_commit>.hml` with the browseable results of the execution.
+   - `<timestamp>-<git_commit>.hml` with the browse-able results of the execution.
```
Comment (leolara) [inline testool/README.md:98]: @adria0 I think the previous form is the correct one.

File: testool/README.md [hunk 2]
```diff
@@ -72,9 +72,9 @@ In the config file you define `[[suite]]`s that defines how tests will be execut
 - `max_steps` the maximum number of executed opcodes. If this is reached, the test is marked to be ignored.
 - `max_gas` the maximum gas of a test. If the is reached, the test is marked to be ignored.
 - `unimplemented_opcodes`, if any of these opcodes are found in the execution trace, the test is marked to be ignored.
```
Comment (leolara) [inline testool/README.md:74]: @adria0 should we remove this documentation about the deleted `unimplemented_opcodes`? or it has not been deleted?

**Comment by leolara** [comment]
LGTM! Left a few optional comments

**Comment by leolara** [comment]
@adria0 perhaps we should remove StateTestError::SkipUnimplemented ?

File: testool/README.md [hunk 3]
```diff
@@ -95,16 +95,19 @@ Sometimes there are some files or specific tests that we want to disable at all.
...
+
 ### Manually executing the tests
 
 Usually we have to debug and run the tests manually to check if everything works ok. We provide a set of command line parameters to help with this.
 
-- `testool [--suite xxx] --cache` to execute all tests, and keeping the results (cache) CSV file. If you delete entries from the cache file, and re-run the tool again, only the deleted tests will be executed again
+- `testool [--suite xxx] --cache <cache_file>` to execute all tests, and keeping the results (cache) CSV file. If you delete entries from the cache file, and re-run the tool again, only the deleted tests will be executed again
 
-- `testool [--suite xxx] --inspect <test_id>` only executed the selected test (even if cached, or ignored). Use `RUST_BACKTRACE=1` here to check if anything fails. Also gives a dumop of the test as also to the geth steps executed.
+- `testool [--suite xxx] --inspect <test_id>` only executed the selected test (even if cached, or ignored). Use `RUST_BACKTRACE=1` here to check if anything fails. Also gives a dump of the test as also to the geth steps executed.
```
[the remark this answers is not in the document]
Comment (leolara) [reply testool/README.md:112]: I think I would rephrase the doc, explaining what you just said. Like, the main thing is that it gives much more info, not just executing the test. Also, "executed -> execute".

[the remark this answers is not in the document]
Comment (leolara) [reply testool/README.md:112]: Perhaps, the best design would be to have a `--tests` flag that only executes the tests in the command, and a `--inspect` arg without parameters that shouls the extra info. But I think that would be another task.

File: testool/README.md [hunk 4]
```diff
@@ -70,11 +70,10 @@ In the config file you define `[[suite]]`s that defines how tests will be execut
 
 - `id` is the identifier of the suite. The default suite is called `default`.
 - `max_steps` the maximum number of executed opcodes. If this is reached, the test is marked to be ignored.
-- `max_gas` the maximum gas of a test. If the is reached, the test is marked to be ignored.
-- `unimplemented_opcodes`, if any of these opcodes are found in the execution trace, the test is marked to be ignored.
-- you should define also only one of these paramers:
-   - `allow_tests` with the list of tests or testsets to execute. All others will be excluded. Test sets should be prefixed with `&`
-   - `ignore_tests` with the list of test or testsets to ignore. All others will be included. Test sets should be prefixed with `&`
+- `max_gas` the maximum gas of a test. If the is reached, the test is marked to be ignored. Put a `0` if you do not want to limit it.
```
Comment (ed255) [inline testool/README.md:73]:
```suggestion
- `max_gas` the maximum gas of a test. If the specified maximum gas is reached, the test is marked to be ignored. Put a `0` if you do not want to limit it.
```

**Review by ed255 — APPROVED** [review]
Just found a phrase missing some content, but otherwise LGTM! Nice clean up in `testool/Config.toml` :D
